src/ui/services/home.ui-service.ts

```diff
@@ -0,0 +1,30 @@
+import { Page } from "@playwright/test";
+import { ModuleName } from "types/home.types";
+import { CustomersPage } from "ui/pages/customers/customers.page";
+import { HomePage } from "ui/pages/home.page";
+import { ProductsPage } from "ui/pages/products/products.page";
+
+export class HomeUIService {
+    homePage: HomePage;
+    customersPage: CustomersPage;
+    productsPage: ProductsPage;
+    constructor(private page: Page) {
+        this.customersPage = new CustomersPage(page);
+        this.homePage = new HomePage(page);
+        this.productsPage = new ProductsPage(page);
+    }
+
+    async openModule(moduleName: ModuleName) {
+        await this.homePage.clickModuleButton(moduleName);
+        switch (moduleName) {
+            case "Customers":
+                await this.customersPage.waitForOpened();
+                break;
+        }
+        switch (moduleName) {
+            case "Products":
+                await this.productsPage.waitForOpened();
+                break;
+        }
+    }
```

Comment:
Если использовать `switch`, то одного достаточно
```suggestion
    async openModule(moduleName: ModuleName) {
        await this.homePage.clickModuleButton(moduleName);
        switch (moduleName) {
            case "Customers":
                await this.customersPage.waitForOpened();
                break;
             case "Products":
                await this.productsPage.waitForOpened();
                break;
        }
    }
```

Альтернатива без `switch`

```ts
 async openModule(moduleName: ModuleName) {
        await this.homePage.clickModuleButton(moduleName);
        await this[`${moduleName.toLowerCase()}Page`].waitForOpened();
}
```